src/components/ui/swiper.tsx

```diff
@@ -0,0 +1,156 @@
+      className={cn("px-2", className)}
+      {...props}
+    >
+      {children}
+    </Button>
+  )
+)
+SwiperNavButton.displayName = "SwiperNavButton"
+
+const SwiperNextButton = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ children, className, ...props }, ref) => (
+    <SwiperNavButton
+      ref={ref}
+      className={cn("ui-swiper-button-next", className)}
+      {...props}
+    >
+      {children || <ChevronNext />}
+    </SwiperNavButton>
+  )
+)
+SwiperNextButton.displayName = "SwiperNextButton"
+
+const SwiperPrevButton = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ children, className, ...props }, ref) => (
+    <SwiperNavButton
+      ref={ref}
+      className={cn("ui-swiper-button-prev", className)}
+      {...props}
+    >
+      {children || <ChevronPrev />}
+    </SwiperNavButton>
+  )
+)
+SwiperPrevButton.displayName = "SwiperPrevButton"
+
+const SwiperPaginationDots = React.forwardRef<
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement>
+>(({ className, ...props }, ref) => (
+  <div
+    ref={ref}
+    className={cn(
+      "ui-swiper-pagination flex w-fit flex-row",
+      "[&_.swiper-pagination-bullet]:bg-primary-high-contrast",
+      "[&_.swiper-pagination-bullet-active]:bg-primary-hover",
+      className
+    )}
+    {...props}
+  />
+))
+SwiperPaginationDots.displayName = "SwiperPaginationDots"
+
+const SwiperNavContainer = React.forwardRef<
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement>
+>(({ className, ...props }, ref) => (
+  <div
+    ref={ref}
+    className={cn(
+      "mx-auto mt-6 flex h-6 w-fit items-center gap-4 rounded-full bg-background-highlight",
+      className
+    )}
+    {...props}
+  />
+))
+SwiperNavContainer.displayName = "SwiperNavContainer"
+
+type SwiperProps = SwiperReactProps & {
+  navigation?: React.ReactNode
+  children: React.ReactNode[]
+}
+const Swiper = React.forwardRef<SwiperRef, SwiperProps>(
+  ({ children, navigation, ...props }, ref) => {
+    const { t } = useTranslation("common")
+    return (
+      <SwiperReact
+        ref={ref}
+        navigation={{
+          nextEl: ".ui-swiper-button-next",
+          prevEl: ".ui-swiper-button-prev",
+        }}
+        a11y={{
+          prevSlideMessage: t("previous"),
+          nextSlideMessage: t("next"),
+        }}
+        pagination={{
+          clickable: true,
+          el: ".ui-swiper-pagination",
+        }}
+        keyboard
+        modules={[Navigation, Pagination, Keyboard, EffectCards]}
+        slidesPerView={1}
+        slidesPerGroup={1}
+        lazyPreloadPrevNext={0}
+        slideClass="swiper-slide"
+        {...props}
+      >
+        {children.map((child, index) => (
+          <SwiperSlide key={index}>{child}</SwiperSlide>
+        ))}
+        {navigation || (
+          <SwiperNavContainer>
+            <SwiperPrevButton />
+            <SwiperPaginationDots />
+            <SwiperNextButton />
+          </SwiperNavContainer>
+        )}
```

Comment:
This is good but I think it is reducing flexibility and customization.

1. To simplify, you could export a `SwiperNavigation` component if wrap this common shape.
2. One goal of exporting all these sub components is to avoid having props like `navigation` because you are reducing flexibility. What if I want the nav to be at the top of the slides and not the bottom? exporting the necessary sub components and rendering just the children gives you all this flexibility for free.

```tsx
// following the convention you could have something like this without any config or custom props
<Swiper>
  <SwiperNavigation />

  <div className="p-4 bg-green-500">
    <SwiperSlide>
      <div className="bg-red-500"></div>
    </SwiperSlide>
  </div>

  <div className="p-4 bg-green-500">
    <SwiperSlide>
      <div className="bg-blue-500"></div>
    </SwiperSlide>
  </div>

  <SwiperNavigation />
</Swiper>
```

Reply:
Yeah, couldn't agree more! implementing these suggestions

Reply:
Just pushed those updates. With how SwiperJS works, it's best to have a container to handle it's constraints (the `height` is a must at a minimum)

So we'll have this structure:

```tsx
<SwiperContainer className="tw-styles-as-needed">
  <Swiper {...swiperOptionsAsNeeded}>
    <SwiperSlide>
      <!-- contents, styling -->
    </SwiperSlide>
    <!-- repeat slides to desires -->
    <SwiperNavigation className="tw-styles-as-needed" />
  </Swiper>
</SwiperContainer>
```

for example:

```tsx
<SwiperContainer>
  <Swiper>
    <SwiperSlide>
      <div className="bg-black">
        <p className="text-white">Hello</p>
      </div>
    </SwiperSlide>
    <SwiperSlide>
      <div className="bg-white">
        <p className="text-black">World</p>
      </div>
    </SwiperSlide>
    <SwiperNavigation />
  </Swiper>
</SwiperContainer>
```

This allows option to not add navigation, or to override styling of each of these using TW classNames. Unfortunately there are limitation to Swiper React that limit our ability to manipulate the `.swiper-wrapper` element that is added by this plugin with JS, and then forces the order of the DOM to have the `SwiperNavigation` elements after it.

As a workaround, I've added a `top` and `bottom` placement variant called `navigationPlacement` to allow user to effectively switch this to `flex flex-col-reverse` to place the nav contents above the swiper slides.